src/main/java/org/mockito/internal/matchers/Equals.java

```diff
@@ -51,12 +51,18 @@ public int hashCode() {
     }
 
     @Override
-    public String toStringWithType() {
-        return "(" + wanted.getClass().getSimpleName() + ") " + describe(wanted);
+    public String toStringWithType(boolean useFullyQualifiedClassName) {
```

Comment:
Rather than a `boolean`, can we pass in the `String` here?

Reply:
Had kept boolean thinking, we either need fully name of simple name, but keeping it as String gives flexibility to the caller to choose the name.